### PR TITLE
Add support for Ubuntu 22.04

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Add support of the latest Ubuntu 22.04.

* [x] Remove specific code for automatically installing optional dependencies and creating packages on Ubuntu 18.04.
* [x] Remove code to compile Python3.6 API (not available by default on Ubuntu 22.04)
* [x] On Ubuntu 22.04, download OpenSSL dependencies from Cyberbotics server
* [x] Create and upload new sumo package for Ubuntu 20.04
* [x] Update instructions to create packages on Ubuntu 22.04
* [x] Update Ubuntu 20.04 package creation to make them work on Ubuntu 22.04 too

Notes:
* ROS is currently not officially supported on Ubuntu 22.04 and it is not known if it will be in the future given that *ROS noetic* is the last release.

Errors fixed:
* ``Warning: Incompatible version of OpenSSL (built with OpenSSL 1.x, runtime version is >= 3.x)``
   -> Default SSL library is version 3 but we Qt6 is still built with version 1.
   We need to re-add the download of default Ubuntu 20.04 system OpenSSL libraries.
* Sumo interface: changed default version of `libgdal` and `libproj`

Tests:
* run Ubuntu 20.04 packages:
    * [x] Debian package
    * [x] tarball -> works after including openssl libraries and additional sumo libraries in the packages.
* local package creation
    * [x] tarball

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
It remains just one issue with the Ubuntu 22.04 tests on the CI where the robot window test fails.
Note that the Ubuntu 22.04 environment is still in beta and for example it seems that Firefox is not installed yet.

I tried to force using chromium instead but it didn't solve the issue.

I probably think it is better to wait until the Ubuntu 22.04 environment for the GitHub action is fully released before further inspecting the issue. For the moment we could simply skip the robot window test and run all the others.

**Comment by stefaniapedrazzi** [comment]
> After cleansing and recompiling the develop branch I noticed that python 3.10 throws a deprecation warning the first time around:
> 
> ```
> controller310.cpp: In function ‘PyObject* PyInit__controller()’:
> controller310.cpp:980:65: warning: ‘void PyEval_InitThreads()’ is deprecated [-Wdeprecated-declarations]
>   980 | #     define SWIG_PYTHON_INITIALIZE_THREADS  PyEval_InitThreads()
>       |                                                                 ^
> controller310.cpp:27431:3: note: in expansion of macro ‘SWIG_PYTHON_INITIALIZE_THREADS’
> 27431 |   SWIG_PYTHON_INITIALIZE_THREADS;
>       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> In file included from /usr/include/python3.10/Python.h:130,
>                  from controller310.cpp:182:
> /usr/include/python3.10/ceval.h:122:37: note: declared here
>   122 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
>       |                                     ^~~~~~~~~~~~~~~~~~
> controller310.cpp:980:65: warning: ‘void PyEval_InitThreads()’ is deprecated [-Wdeprecated-declarations]
>   980 | #     define SWIG_PYTHON_INITIALIZE_THREADS  PyEval_InitThreads()
>       |                                                                 ^
> controller310.cpp:27431:3: note: in expansion of macro ‘SWIG_PYTHON_INITIALIZE_THREADS’
> 27431 |   SWIG_PYTHON_INITIALIZE_THREADS;
>       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> In file included from /usr/include/python3.10/Python.h:130,
>                  from controller310.cpp:182:
> /usr/include/python3.10/ceval.h:122:37: note: declared here
>   122 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
> ```
> 
> Is it being addressed in this PR already?

Yes. It is fixed in this PR.
There are a couple of other fixes related to Python 3.10 that I will also port to the master branch.

File: scripts/install/linux_compilation_dependencies.sh [hunk 1]
```diff
@@ -11,6 +11,8 @@ apt install --yes git lsb-release cmake swig libglu1-mesa-dev libglib2.0-dev lib
 UBUNTU_VERSION=$(lsb_release -rs)
 if [[ $UBUNTU_VERSION == "20.04" ]]; then
        apt install --yes libzip5
+elif [[ $UBUNTU_VERSION == "22.04" ]]; then
+       apt install --yes libzip4
```
Comment (ad-daniel) [inline scripts/install/linux_compilation_dependencies.sh:15]: Is it normal we install libzip4 on 22, and libzip5 on 20?

**Review by ad-daniel — CHANGES_REQUESTED** [review]
I've tested the installation procedure on a virtualbox and it appears to be working nicely, SUMO included. Mostly very minor details/questions. Thank you

File: docs/guide/installation-procedure.md [hunk 2]
```diff
@@ -142,7 +141,7 @@ However, the sand-boxing constraints of snaps yield the following limitations:
 ##### Download Size
 
 The download is significantly bigger as it includes all the dependencies of Webots (ffmpeg, Python, C++ and Java compilers, etc.).
-For Webots R2019b revision 1, the download size of the snap is 1.8GB compared to 1.3GB of the Debian and tarball packages.
+For Webots R2022a, the download size of the snap is 821MB compared to 290MB of the Debian and tarball packages.
```
[the remark this answers is not in the document]
Comment (stefaniapedrazzi) [reply docs/guide/installation-procedure.md:144]: Yes, but we don't know exactly which size the package will be yet.
And we are not even interested in updating this information for every release.
I think it is better to use exact old data than data related to R2022b that we will have to update again later or will be wrong.

File: scripts/install/linux_compilation_dependencies.sh (hunk 1 above)
Comment (stefaniapedrazzi) [reply scripts/install/linux_compilation_dependencies.sh:15]: Yes, this is the default library in Ubuntu 22.04.

File: docs/guide/system-requirements.md [hunk 3]
```diff
@@ -13,11 +13,11 @@ Webots works well on all the graphics cards included in fairly recent Apple comp
...
 
-- Linux: Webots is ensured to run on the latest Ubuntu Long Term Support (LTS) release, currently version 20.04.
+- Linux: Webots is ensured to run on the latest Ubuntu Long Term Support (LTS) release, currently version 22.04.
 But it is also known to run on most recent major Linux distributions, including RedHat, Mandrake, Debian, Gentoo, Arch, SuSE, and Slackware.
 We recommend using a recent version of Linux.
 Webots is provided for Linux 64 (x86-64) systems.
-Webots doesn't run on Ubuntu versions earlier than 18.04.
+Webots doesn't run on Ubuntu versions earlier than 20.04.
 - Windows: Webots runs on Windows 10 and Windows 8.1 (64-bit versions only).
 - Mac: Webots runs on macOS 10.15 "Catalina" and 10.14 "Mojave".
```
[the remark this answers is not in the document]
Comment (stefaniapedrazzi) [reply docs/guide/system-requirements.md:22]: I would update this in a different PR.

**Comment by stefaniapedrazzi** [comment]
I think I addressed all the comments. I would update the documentation of supported macOS and Windows version in a different PR, if it is the case we should also change it in the master/released branch.

**Review by ad-daniel — APPROVED** [review]
Thank you

File: scripts/packaging/Makefile [hunk 4]
```diff
@@ -106,7 +106,7 @@ install: all
 
 $(WEBOTS_DISTRO_EXE): webots_distro.c
 	+@echo "# compiling webots_distro.c"
-	+@$(CC) $(CFLAGS) -Wall -o $@ webots_distro.c $(LIBRARIES)
+	+@$(CC) $(CFLAGS) -Wno-deprecated-declarations -Wall -o $@ webots_distro.c $(LIBRARIES)
```
Comment (omichel) [inline scripts/packaging/Makefile:109]: I assume that deprecated declarations silenced here are specific to Ubuntu 22.04. If so, I would put this inside the Ubuntu 20.04 `CFLAGS`.

File: tests/test_suite.py [hunk 5]
```diff
@@ -30,6 +30,11 @@
 
 from command import Command
 
+is_ubuntu_22_04 = False
+if sys.platform == 'linux':
+    result = subprocess.run(['lsb_release', '-sr'], stdout=subprocess.PIPE)
+    is_ubuntu_22_04 = result.stdout.decode().strip() == '22.04'
+
```
Comment (omichel) [inline tests/test_suite.py:37]: This is just a minor style issue: there is a double initialization here, instead we should set `is_ubuntu_22_04` to `False` in a `else` statement.

**Pull request merged**